### DISCUSSION TITLE
How to set persistent layout in <script setup>?

### DISCUSSION THREAD

**Discussion by sarukomine** [inertiajs/inertia#1164] [Help (Vue)]
how to set persistent layout in `<script setup>`?

```javascript
<script setup>
  import Layout from './Layout'
  useLayout(Layout) // Something like that?
</script>
```

**Comment by julianstark999** [comment]
You could use this plugin [sxzz/unplugin-vue-define-options](https://github.com/sxzz/unplugin-vue-define-options). After you installed it, you can use `defineOptions` in the script to define a layout:

```vue
<script setup>
import AppLayout from '@/Layouts/AppLayout',

defineOptions({
    layout: AppLayout
});
</script>
```

**Reply by madsh93** [reply]
This currently the best way to manually set layouts using the <script setup>.

If however, you want to set the layout automatically by rules, check out this code by Jonathan Reinink (note: this is using Vite)

![image](https://user-images.githubusercontent.com/8860723/181697234-6db8136e-31cd-427a-ac4f-3c89f6353b42.jpeg)
